modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java

```diff
@@ -350,6 +375,9 @@ public String getSchemaType(Schema p) {
 
         if (ref != null && !ref.isEmpty()) {
             type = openAPIType;
+        } else if (openAPIType == "object" && ModelUtils.isAnyTypeSchema(p)) {
```

Comment:
Should this be `"object".equals(openAPIType)` ?

Reply:
yes, thank you. Resolved